lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -384,6 +398,34 @@ data ApiAddress (n :: NetworkDiscriminant) = ApiAddress
     , state :: !(ApiT AddressState)
     } deriving (Eq, Generic, Show)
 
+newtype ApiScript = ApiScript
+    { script :: ApiT Script
```

Comment:
I think if the sole field is going to be `ApiT Script`, then you don't need the `ApiScript` newtype.

Reply:
Perhaps the newtype might be needed if you want automatic JSON instances, but it looks like they are non-generically defined below.